src/s3/util/fileViewerManager.ts

```diff
@@ -55,13 +65,32 @@ export class S3FileViewerManager {
         }
         getLogger().verbose(`S3FileViewer: File from s3 or temp to be opened is: ${fileLocation}`)
 
-        //TODOD:: delegate this logic to S3Tab.ts
-        //this will display the document at the end
-        this.window.showTextDocument(fileLocation)
+        const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation)
+        await newTab.openFileOnReadOnly()
+
+        this.activeTabs.set(fileLocation.fsPath, newTab)
+    }
+
+    public async openOnEditMode(uriOrNode: vscode.Uri | S3FileNode): Promise<void> {
+        if (uriOrNode instanceof vscode.Uri) {
+            //was activated from an open tab
+            if (this.activeTabs.has(uriOrNode.fsPath)) {
+                const tab = this.activeTabs.get(uriOrNode.fsPath)
+                await tab!.openFileOnEditMode()
+            }
+        } else if (uriOrNode instanceof S3FileNode) {
+            //was activated from the explorer, need to get the file
+            const fileLocation = await this.getFile(uriOrNode)
+            if (!fileLocation) {
+                return
+            }
+            const newTab = this.activeTabs.get(fileLocation.fsPath) ?? new S3Tab(fileLocation)
+            await newTab.openFileOnEditMode()
+        }
     }
 
     /**
-     * Fetches a file from S3 or gets it from the local cache if possible.
+     * Fetches a file from S3 or gets it from the local cache if possible and still valid.
```

Comment:
What does "Valid" mean in this case? Is there a reason it might become invalid?

Reply:
Provided in `checkForValidity()` in the next PR. Also elaborated a little more on this documentation.